### PR TITLE
Remove a typedef thats unused when compiling in release mode.

### LINKED ISSUE #5840: Fix warnings for unused local typedef
The following warnings appear in release mode:
```
./source/hp/dof_handler.cc:602:44: warning: unused typedef 'BaseClass' [-Wunused-local-typedef]
          typedef DoFHandler<dim,spacedim> BaseClass;
                                           ^
./source/hp/dof_handler.cc:628:44: warning: unused typedef 'BaseClass' [-Wunused-local-typedef]
          typedef DoFHandler<dim,spacedim> BaseClass;
                                           ^
./source/hp/dof_handler.cc:658:44: warning: unused typedef 'BaseClass' [-Wunused-local-typedef]
          typedef DoFHandler<dim,spacedim> BaseClass;
                                           ^
```

### PR DESCRIPTION
**Pull request by jppelteret**
Fixes #5840.

### CODE REVIEW

File: source/hp/dof_handler.cc [hunk 1]
```diff
@@ -599,12 +599,10 @@ namespace internal
         {
           const unsigned int dim = 1;
 
-          typedef DoFHandler<dim,spacedim> BaseClass;
-
           Assert (dof_handler.finite_elements != nullptr,
-                  typename BaseClass::ExcNoFESelected());
+                  typename DoFHandler<dim,spacedim>::ExcNoFESelected());
```
Comment (bangerth) [inline source/hp/dof_handler.cc]: Nope, not allowed -- the preprocessor will complain about there being 3 arguments to `Assert` separated by commas. You'll have to enclose the whole second argument in parentheses for this to work. (This is why the code introduces the `typedef` in the first place.)

Comment (jppelteret) [reply source/hp/dof_handler.cc]: Thankfully adding some parentheses doesn't take too much effort :-)

**Comment by bangerth** [comment]
OK to merge once the tester is happy.

**Pull request merged**